e2e/tests/test_53_search_filter.py

```diff
@@ -0,0 +1,67 @@
+from .fixtures.hub import AuthHeaders, AuthType
+from .utils import get_default_config_split, poll_until_ready_and_assert
+
+
+def test_filter_endpoint(
+    auth_headers: AuthHeaders,
+    hf_public_dataset_repo_csv_data: str,
+) -> None:
+    auth: AuthType = "none"
+    expected_status_code: int = 200
+    expected_error_code = None
+    # TODO: add dataset with various splits, or various configs
+    dataset = hf_public_dataset_repo_csv_data
+    config, split = get_default_config_split()
+    headers = auth_headers[auth]
+    offset = 1
+    length = 2
+    where = "col_4 = 'B'"
+    filter_response = poll_until_ready_and_assert(
+        relative_url=(
+            f"/filter?dataset={dataset}&config={config}&split={split}&offset={offset}&length={length}&where={where}"
+        ),
+        expected_status_code=expected_status_code,
+        expected_error_code=expected_error_code,
+        headers=headers,
+        check_x_revision=True,
+    )
+    if not expected_error_code:
```

Comment:
I think we can remove this validation since we know expected_error_code is always None.
I saw same code in test_rows_endpoint and test_search_endpoint  :confused:

Reply:
Done.